src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -39,23 +38,53 @@ Chunk ORCBlockInputFormat::generate()
     if (!file_reader)
         prepareReader();
 
-    if (stripe_current >= stripe_total)
-        return res;
+    while (true)
```

Comment:
Not important, but `generate` method now a bit difficult to read, maybe it would be better to split it into some small methods like `getNextBatchReader`, `getNextBatch`. If you don't think so or don't have time, just ignore it